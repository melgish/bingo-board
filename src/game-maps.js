export const coverAll = [
  [1, 1, 1, 1, 1],
  [1, 1, 1, 1, 1],
  [1, 1, 1, 1, 1],
  [1, 1, 1, 1, 1],
  [1, 1, 1, 1, 1],
]

export const bigBox = [
  [1, 1, 1, 1, 1],
  [1, 0, 0, 0, 1],
  [1, 0, 0, 0, 1],
  [1, 0, 0, 0, 1],
  [1, 1, 1, 1, 1],
]

export const smallBox = [
  [0, 0, 0, 0, 0],
  [0, 1, 1, 1, 0],
  [0, 1, 0, 1, 0],
  [0, 1, 1, 1, 0],
  [0, 0, 0, 0, 0],
]

export const bigCorners = [
  [1, 1, 0, 1, 1],
  [1, 1, 0, 1, 1],
  [0, 0, 0, 0, 0],
  [1, 1, 0, 1, 1],
  [1, 1, 0, 1, 1],
]

export const letterX = [
  [1, 0, 0, 0, 1],
  [0, 1, 0, 1, 0],
  [0, 0, 1, 0, 0],
  [0, 1, 0, 1, 0],
  [1, 0, 0, 0, 1],
]

export const letterT = [
  [1, 1, 1, 1, 1],
  [0, 0, 1, 0, 0],
  [0, 0, 1, 0, 0],
  [0, 0, 1, 0, 0],
  [0, 0, 1, 0, 0],
]

export const cross = [
  [0, 0, 1, 0, 0],
  [0, 0, 1, 0, 0],
  [1, 1, 1, 1, 1],
  [0, 0, 1, 0, 0],
  [0, 0, 1, 0, 0],
]

export const letterL = [
  [1, 0, 0, 0, 0],
  [1, 0, 0, 0, 0],
  [1, 0, 0, 0, 0],
  [1, 0, 0, 0, 0],
  [1, 1, 1, 1, 1],
]

export const normals = [
  [
    [1, 1, 1, 1, 1],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
  ],
  [
    [0, 0, 0, 0, 0],
    [1, 1, 1, 1, 1],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
  ],
  [
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [1, 1, 1, 1, 1],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
  ],
  [
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [1, 1, 1, 1, 1],
    [0, 0, 0, 0, 0],
  ],
  [
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [1, 1, 1, 1, 1],
  ],
  [
    [1, 0, 0, 0, 0],
    [1, 0, 0, 0, 0],
    [1, 0, 0, 0, 0],
    [1, 0, 0, 0, 0],
    [1, 0, 0, 0, 0],
  ],
  [
    [0, 1, 0, 0, 0],
    [0, 1, 0, 0, 0],
    [0, 1, 0, 0, 0],
    [0, 1, 0, 0, 0],
    [0, 1, 0, 0, 0],
  ],
  [
    [0, 0, 1, 0, 0],
    [0, 0, 1, 0, 0],
    [0, 0, 1, 0, 0],
    [0, 0, 1, 0, 0],
    [0, 0, 1, 0, 0],
  ],
  [
    [0, 0, 0, 1, 0],
    [0, 0, 0, 1, 0],
    [0, 0, 0, 1, 0],
    [0, 0, 0, 1, 0],
    [0, 0, 0, 1, 0],
  ],
  [
    [0, 0, 0, 0, 1],
    [0, 0, 0, 0, 1],
    [0, 0, 0, 0, 1],
    [0, 0, 0, 0, 1],
    [0, 0, 0, 0, 1],
  ],
  [
    [1, 0, 0, 0, 0],
    [0, 1, 0, 0, 0],
    [0, 0, 1, 0, 0],
    [0, 0, 0, 1, 0],
    [0, 0, 0, 0, 1],
  ],
  [
    [0, 0, 0, 0, 1],
    [0, 0, 0, 1, 0],
    [0, 0, 1, 0, 0],
    [0, 1, 0, 0, 0],
    [1, 0, 0, 0, 0],
  ],
]

export const GAME_MAPS = [
  bigBox,
  smallBox,
  bigCorners,
  letterX,
  letterT,
  cross,
  letterL,
  coverAll,
]
