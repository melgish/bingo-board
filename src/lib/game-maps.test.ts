import { describe, expect, it } from "vitest"
import { GAME_MAPS } from "./game-maps"

describe("GAME_MAPS", () => {
  it("should represent games", () => {
    expect(GAME_MAPS).toMatchSnapshot()
  })
})
