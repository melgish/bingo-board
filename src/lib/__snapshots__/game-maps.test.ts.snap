// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GAME_MAPS > should represent games 1`] = `
[
  [
    [
      1,
      1,
      1,
      1,
      1,
    ],
    [
      1,
      0,
      0,
      0,
      1,
    ],
    [
      1,
      0,
      0,
      0,
      1,
    ],
    [
      1,
      0,
      0,
      0,
      1,
    ],
    [
      1,
      1,
      1,
      1,
      1,
    ],
  ],
  [
    [
      0,
      0,
      0,
      0,
      0,
    ],
    [
      0,
      1,
      1,
      1,
      0,
    ],
    [
      0,
      1,
      0,
      1,
      0,
    ],
    [
      0,
      1,
      1,
      1,
      0,
    ],
    [
      0,
      0,
      0,
      0,
      0,
    ],
  ],
  [
    [
      1,
      1,
      0,
      1,
      1,
    ],
    [
      1,
      1,
      0,
      1,
      1,
    ],
    [
      0,
      0,
      0,
      0,
      0,
    ],
    [
      1,
      1,
      0,
      1,
      1,
    ],
    [
      1,
      1,
      0,
      1,
      1,
    ],
  ],
  [
    [
      1,
      0,
      0,
      0,
      1,
    ],
    [
      0,
      1,
      0,
      1,
      0,
    ],
    [
      0,
      0,
      1,
      0,
      0,
    ],
    [
      0,
      1,
      0,
      1,
      0,
    ],
    [
      1,
      0,
      0,
      0,
      1,
    ],
  ],
  [
    [
      1,
      1,
      1,
      1,
      1,
    ],
    [
      0,
      0,
      1,
      0,
      0,
    ],
    [
      0,
      0,
      1,
      0,
      0,
    ],
    [
      0,
      0,
      1,
      0,
      0,
    ],
    [
      0,
      0,
      1,
      0,
      0,
    ],
  ],
  [
    [
      0,
      0,
      1,
      0,
      0,
    ],
    [
      0,
      0,
      1,
      0,
      0,
    ],
    [
      1,
      1,
      1,
      1,
      1,
    ],
    [
      0,
      0,
      1,
      0,
      0,
    ],
    [
      0,
      0,
      1,
      0,
      0,
    ],
  ],
  [
    [
      1,
      0,
      0,
      0,
      0,
    ],
    [
      1,
      0,
      0,
      0,
      0,
    ],
    [
      1,
      0,
      0,
      0,
      0,
    ],
    [
      1,
      0,
      0,
      0,
      0,
    ],
    [
      1,
      1,
      1,
      1,
      1,
    ],
  ],
  [
    [
      1,
      1,
      1,
      1,
      1,
    ],
    [
      1,
      1,
      1,
      1,
      1,
    ],
    [
      1,
      1,
      1,
      1,
      1,
    ],
    [
      1,
      1,
      1,
      1,
      1,
    ],
    [
      1,
      1,
      1,
      1,
      1,
    ],
  ],
]
`;
